tests/ttnn/unit_tests/operations/test_moreh_softmin.py

```diff
@@ -397,3 +388,91 @@ def test_softmin_backward_optional_output_tensor(shape_dim, optional_output_tens
     passing, out = comp_allclose_and_pcc(x.grad, tt_dev, rtol=rtol, atol=atol)
     logger.info(out)
     assert passing
+
+
+@pytest.mark.parametrize(
+    "shape_dim",
+    [
+        [[32, 32], 1],  # single tile
+        [[3, 32, 32 * 5], 2],  # mutiple tile with dim W
+        [[5, 6, 32, 32], 3],  # multiple cores
+        [[10, 20, 32 * 3, 32 * 5], 3],  # multiple tiles per core
+        [[32, 32], 0],  # single tile
+        [[3, 32 * 5, 32], 1],  # mutiple tile with dim H
+        [[5, 6, 32, 32], 2],  # multiple cores
+        [[10, 20, 32 * 3, 32 * 5], 2],  # multiple tiles per core
+    ],
+)
+@pytest.mark.parametrize("compute_kernel_options", compute_kernel_options, ids=compute_kernel_ids)
+def test_softmin_callback(shape_dim, compute_kernel_options, device, use_program_cache):
+    shape, dim = shape_dim
+    torch.manual_seed(0)
+
+    compute_kernel_config = get_compute_kernel_options(compute_kernel_options)
+
+    x = torch.randint(low=0, high=4, size=shape).to(torch.bfloat16)
+
+    dev_x = ttnn.Tensor(x, ttnn.bfloat16).to(ttnn.TILE_LAYOUT).to(device)
+
+    tt_cpu = F.softmin(x, dim)
+    for i in range(2):
+        tt_npu = ttnn.operations.moreh.softmin(dev_x, dim, compute_kernel_config=compute_kernel_config)
+        if i == 0:
+            num_program_cache_entries = device.num_program_cache_entries()
+            assert num_program_cache_entries > 0
+        else:
+            assert device.num_program_cache_entries() == num_program_cache_entries
```

Comment:
```Python
    torch.manual_seed(2024)
    for _ in range(2):
        run_getitem_RAW_MAJOR(shape_index_dim, dtype, index_size, device)
        torch_dummy = torch.randn([32, 32])
        tt_dummy = to_npu(torch_dummy, device)
```
https://github.com/tenstorrent/tt-metal/blob/main/tests/ttnn/unit_tests/operations/test_moreh_getitem.py#L277

It seems that the current method doesn't change either the `value` or the `address` of the tensor between the two executions. In this case, the test may pass without updating the `address` in the callback implementation. 

Like the getitem example above, we should modify the tensor `value` and add a dummy tensor to ensure the `address` changes.